subprojects/smoke-ide-test/build.gradle.kts

```diff
@@ -0,0 +1,46 @@
+import gradlebuild.integrationtests.tasks.SmokeIdeTest
+import gradlebuild.integrationtests.addDependenciesAndConfigurations
+import gradlebuild.integrationtests.ide.AndroidStudioProvisioningExtension
+
+plugins {
+    id("gradlebuild.internal.java")
+    id("gradlebuild.android-studio-provisioning")
+}
+
+description = "Tests are checking Gradle behavior during IDE synchronization process"
+
+val smokeIdeTestSourceSet = sourceSets.create("smokeIdeTest") {
+    compileClasspath += sourceSets.main.get().output
+    runtimeClasspath += sourceSets.main.get().output
+}
+
+addDependenciesAndConfigurations("smokeIde")
+val smokeIdeTestImplementation: Configuration by configurations
+val smokeIdeTestDistributionRuntimeOnly: Configuration by configurations
+
+plugins.withType<IdeaPlugin> {
+    with(model) {
+        module {
+            testSources.from(smokeIdeTestSourceSet.java.srcDirs, smokeIdeTestSourceSet.groovy.srcDirs)
+            testResources.from(smokeIdeTestSourceSet.resources.srcDirs)
+        }
+    }
+}
+
+val jvmArgumentProvider = the<AndroidStudioProvisioningExtension>().androidStudioSystemProperties(project, emptyList())
```

Comment:
🚀 Move it to the task configuration block so it's only computed when the task is part of the work graph